submissions/evgeniy24/friends_app/app.js

```diff
@@ -0,0 +1,248 @@
+            return friend.name.first.toLowerCase().includes(searchName)
+        })
+        return renderFriendsList(searchResultList);
+    } else {
+        searchResultList = [...friendsList]
+        renderFriendsList(friendsList);
+    }
+})
+
+SORTING_MENU.addEventListener('click', function (event) {
+    if (filteredByGenderList) {
+        sortedList = filteredByGenderList;
+    } else if (searchResultList) {
+        sortedList = searchResultList;
+    } else {
+        sortedList = [...friendsList];
+    }
+
+    const searchByName = function (a, b) {
+        const nameA = a.name.first.toUpperCase();
+        const nameB = b.name.first.toUpperCase();
+        return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+    }
+
+    const sortByAge = function (a, b) {
+        return a.dob.age - b.dob.age;
+    }
+
+    if (event.target.value === 'az') {
+        sortedList.sort(searchByName);
+        renderFriendsList(sortedList);
+
+    } else if (event.target.value === 'za') {
+        sortedList.sort((a, b) => searchByName(b, a));
+        renderFriendsList(sortedList);
+
+    } else if (event.target.value === 'young') {
+        sortedList.sort(sortByAge);
+        renderFriendsList(sortedList);
+
+    } else if (event.target.value === 'older') {
+        sortedList.sort((a, b) => sortByAge(b, a));
+        renderFriendsList(sortedList);
+    }
+})
+
+FILTER_MENU.addEventListener('click', function (event) {
+
+    if (searchResultList) {
+        genderNeutralList = searchResultList;
+    } else {
+        genderNeutralList = [...friendsList];
+    }
+
+    if (event.target.value === 'male') {
+        filteredByGenderList = genderNeutralList.filter(function (friend) {
+            return friend.gender === 'male';
+        })
+        renderFriendsList(filteredByGenderList);
+
+    } else if (event.target.value === 'female') {
+        filteredByGenderList = genderNeutralList.filter(function (friend) {
+            return friend.gender === 'female';
+        })
+        renderFriendsList(filteredByGenderList);
+
+    } else if (event.target.value === 'all') {
+        filteredByGenderList = genderNeutralList;
+        renderFriendsList(filteredByGenderList);
+    }
+})
+
+function renderErrorMessage() {
+    FRIENDS_LIST_WRAP.innerHTML =
+        `
+    <div class="error-message-wrap">
+        <p class="error-msg-text">OOPS! Something Bad Happened, please try again</p>
+        <button class="radio__button error-button" value="loading-error">Try Again</button>
+    </div>
+    `
+}
+
+MOBILE_MENU_BUTTON.addEventListener('click', function () {
+    clickMenuBtn(MOBILE_MENU_BUTTON);
+    clickMenuBtn(MENU_WRAP);
+    scrollDepth = window.pageYOffset;
+    scrollToTop();
+});
+
+function clickMenuBtn(elem) {
+    elem.classList.toggle('change');
+}
+
+function scrollToTop() {
+    let timer = 0;
+    if (scrollDepth > 100) {
+        window.scrollTo(pageXOffset, scrollDepth);
+        scrollDepth = scrollDepth - 60;
```

Comment:
Why it is 60 and what is 15? [Magic numbers](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad)!

Reply:
Resolved